backend/src/routes/api/cre/creUtils.ts

```diff
@@ -107,6 +108,20 @@ export const postCRE = async (
         name: body.imagePullSecretName,
       };
     }
+  }
+  else if (body.buildType === 'GitRepository') {
+    if (!body.gitRef || !body.repository) {
+      return {
+        success: false,
+        error: "Parameters 'gitRef' and 'repository' are expected when using buildType: 'GitRepository'",
```

Comment:
Actually, `gitRef` is optional:

https://github.com/thoth-station/meteor-operator/blob/825a2519425ada316d545ba8b2b4b60d2168b4bb/api/v1alpha1/customruntimeenvironment_types.go#L71-L73